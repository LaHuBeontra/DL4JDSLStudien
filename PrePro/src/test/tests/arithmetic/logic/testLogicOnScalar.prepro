function main() {
	import scal numbers, scal numbers2;

	scal equal = numbers == numbers2;
	scal less = testLess(numbers, numbers2);
	scal lessOrEqual = numbers <= numbers2;
	scal greater = numbers > numbers2;
	scal greaterOrEqual = numbers >= numbers2;

	export numbers, numbers2, equal, less, lessOrEqual, greater, greaterOrEqual;
}

function testLess(scal scal1, scal scal2) returns scal {
	return scal1 < scal2;
}
