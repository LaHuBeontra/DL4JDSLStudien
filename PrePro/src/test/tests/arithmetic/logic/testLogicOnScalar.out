DefaultPreProDataSet{
dataSetEntries=
[DataSetEntry{variableName='numbers', variableShape=[4, 1], startColumn=0, length=1}, DataSetEntry{variableName='numbers2', variableShape=[4, 1], startColumn=1, length=1}, DataSetEntry{variableName='equal', variableShape=[4, 1], startColumn=2, length=1}, DataSetEntry{variableName='less', variableShape=[4, 1], startColumn=3, length=1}, DataSetEntry{variableName='lessOrEqual', variableShape=[4, 1], startColumn=4, length=1}, DataSetEntry{variableName='greater', variableShape=[4, 1], startColumn=5, length=1}, DataSetEntry{variableName='greaterOrEqual', variableShape=[4, 1], startColumn=6, length=1}],
dataSet=
[[1.00,  11.00,  0.00,  1.00,  1.00,  0.00,  0.00],  
 [2.00,  12.00,  0.00,  1.00,  1.00,  0.00,  0.00],  
 [3.00,  3.00,  1.00,  0.00,  1.00,  0.00,  1.00],  
 [4.00,  -14.00,  0.00,  0.00,  0.00,  1.00,  1.00]]

constants={}

calculated_variables=
equal=[0.00,  0.00,  1.00,  0.00]
greaterOrEqual=[0.00,  0.00,  1.00,  1.00]
numbers2=[11.00,  12.00,  3.00,  -14.00]
numbers=[1.00,  2.00,  3.00,  4.00]
lessOrEqual=[1.00,  1.00,  1.00,  0.00]
less=[1.00,  1.00,  0.00,  0.00]
greater=[0.00,  0.00,  0.00,  1.00]
}